src/avro_ocf.erl

```diff
@@ -253,10 +255,23 @@ decode_block(_Lkup, _Type, _Codec, <<>>, 0, Acc, _Options) -> Acc;
 decode_block(Lkup, Type, deflate, Bin, Count, Acc, Options) ->
   Decompressed = zlib:unzip(Bin),
   decode_block(Lkup, Type, null, Decompressed, Count, Acc, Options);
+decode_block(Lkup, Type, snappy, Bin, Count, Acc, Options) ->
+  Size = byte_size(Bin),
+  <<Compressed:(Size-4)/binary, Checksum:4/binary>> = Bin,
+  {ok, Decompressed} = snappyer:decompress(Compressed),
+  case is_invalid_checksum(Decompressed, Checksum) of
+    false -> Decompressed;
+    true -> erlang:error({invalid_checksum, Decompressed})
+  end,
+  decode_block(Lkup, Type, null, Decompressed, Count, Acc, Options);
 decode_block(Lkup, Type, null, Bin, Count, Acc, Options) ->
   {Obj, Tail} = decode_stream(Lkup, Type, Bin, Options),
   decode_block(Lkup, Type, null, Tail, Count - 1, [Obj | Acc], Options).
 
+- spec is_invalid_checksum(binary(), binary()) -> boolean().
+is_invalid_checksum(Decompressed, Checksum) ->
+  binary:encode_unsigned(erlang:crc32(Decompressed)) /= Checksum.
```

Comment:
nit: as this doesn't involve float/integer conversions I'd prefer exact inequality `=/=` here

Reply:
Changed to `=:=` since I have renamed the function to `is_valid_checksum`